src/org/jetbrains/java/decompiler/struct/StructContext.java

```diff
@@ -165,8 +165,8 @@ public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IO
 
       @Override
       public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
-        String dirStr = dir.toString();
-        if (dirStr.length() > 1) unit.addDirEntry(dirStr.substring(1));
+        String dirStr = dir.toString() + "/"; // We are saving folders, not files here Path#toString() treats the path as a file by default.
+        if (dirStr.length() > 2) unit.addDirEntry(dirStr.substring(1));
```

Comment:
I think that can also be reverted then?

Reply:
it might be better to keep it just in case another implementation of IResultSaver has a similar issue

Reply:
Then that `IResultSaver` is incorrect.

Reply:
The main problem is that `IResultSaver` has no javadoc (so the method names are the only indication of what they should do) and is generally an awful interface, but we're not gonna fix that today.